posthog/src/main/java/com/posthog/PostHogInterface.kt

```diff
@@ -0,0 +1,52 @@
+package com.posthog
+
+public interface PostHogInterface {
+    public fun <T : PostHogConfig> setup(config: T)
+
+    public fun close()
+
+    public fun capture(
+        event: String,
+        distinctId: String? = null,
+        properties: Map<String, Any>? = null,
+        userProperties: Map<String, Any>? = null,
+        userPropertiesSetOnce: Map<String, Any>? = null,
+        groupProperties: Map<String, Any>? = null,
+    )
+
+    public fun identify(
+        distinctId: String,
+        properties: Map<String, Any>? = null,
+        userProperties: Map<String, Any>? = null,
+        userPropertiesSetOnce: Map<String, Any>? = null,
+        // TODO: should we have groupProperties here?
```

Comment:
@pauldambra do you know about that? if the identify also accepts group properties? I found this mixed up in SDKs

Reply:
I'd expect posthog-js to be the canonical implementation right now.

There when you call `posthog.group` we register super properties that are sent with every subsequent event. Confusingly (to me) events have groups but people don't.

Since Android library is stateful then I think it would work the same way.

I guess Java library will be stateless and so we'd either have to register super properties by distinct id or insist that groups are always passed by the caller

Reply:
thanks, removed the comment, I didn't find any code path for posthog-js

Reply:
Yeah this is right, `identify()` will never set group properties, even in stateless SDKs because the group properties have nothing to do with the person identity.

Only `posthog.group()` calls should have the parameter to set group properties :)